### DISCUSSION TITLE
Global Span attribute [Span attributes Readability & Span attributes addition to SpanContext]

### DISCUSSION THREAD

**Discussion by AlexanderArgyriou** [open-telemetry/opentelemetry-java-instrumentation#7396] [Q&A] [answered]
More or less i am trying to achieve something like that https://stackoverflow.com/questions/74761054/opentelemetry-custom-traceid-addition Both requirements and proposed solutions are described in my question. Any ideas, directions, etc are always welcome.

https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14026 describes exactly my case.
This could be a solution, no difference with the scenario of carrying a userId for example. Is this supported by otel somehow ?

Thanks in advance

**Comment by laurit** [comment]
Do I understand correctly that you wish to attach a custom trace id that is passed in the request as an attribute to all the spans?

**Reply by AlexanderArgyriou** [reply]
exactly

**Reply by laurit** [reply]
You could write an agent extension (https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension) That uses a `SpanProcessor`(https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/examples/extension/src/main/java/com/example/javaagent/DemoSpanProcessor.java) to set the attribute. Place your custom trace id in `Context` then in the `SpanProcessor` you can easily access it. The tricky part will be placing the trace id in the context. For this it is probably easiest if you write the code that extracts the trace id in your application, perhaps as a servlet filter or something else that you can place around the request handling. The idea is to write the code in your application so that it would be easy to instrument from your extension so that the extension could capture your custom trace id and place it into the context. You could think that instead of instrumenting you could write the code that places the trace id into context inside your application using opentelemetry apis, but this won't work as you can't easily share a context key between the application and the agent.

**Reply by AlexanderArgyriou** [reply]
Thanks for your effort to explain to me this scenario. I found a suitable way using Baggage api propagation. I wrote a custom span processor so i achieved propagating custom attributes through Baggage.current() and populate them on spans while overriding the onStart() method of span processor.

**Comment by AlexanderArgyriou** [comment, marked as answer]
I found a suitable way using Baggage api propagation. I wrote a custom span processor so i achieved propagating custom attributes through Baggage.current() and populate them on spans while overriding the onStart() method of span processor.

**Reply by ppr1011** [reply]
Hello, I also encountered the problem you mentioned. Can you tell me how you solved it? Is there any reference code, thank you very much
